### PR TITLE
chore(contracts): comment bindings type conflcts

### PR DESCRIPTION
**Pull request by kevinhalliday**
Cleanup bindings, add autocomment.

task: none

### CODE REVIEW

**Comment by kevinhalliday** [comment]
Cleanup: 
- gen bindings for MockERC20, not ERC20
- remove EIP1271SignatureUtils bindings (it's a library)

Add autocomment:
- abigen generators go structs for each solidity struct it encounters in a file, this leads to conflicts
- on latest bindings update, we merged with manual comments
- bindings should not need manual comments, as we will run them in CI

**Review by pavelnikolov — APPROVED** [review]

File: contracts/bindings/scripts/commenttypes.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+// commentTypes comments out type definitions in a Go file.
+func commentTypes(content string, types []string) string {
+	var modifiedLines []string
+	lines := strings.Split(content, "\n")
+
+	// tracks whether we're in a type definition
+	//
+	// type Name {
+	// ...
+	// }
```
Comment (pavelnikolov) [inline contracts/bindings/scripts/commenttypes.go:77]: Ideally, we need to do this using some form of AST.

Comment (kevinhalliday) [reply contracts/bindings/scripts/commenttypes.go:77]: yeah but this is pretty lowstakes, single usecase, so quick and dirty is fine

Comment (pavelnikolov) [reply contracts/bindings/scripts/commenttypes.go:77]: Take a look at [this library](https://github.com/unpackdev/solgo).
Alternatively [this SO answer](https://ethereum.stackexchange.com/questions/28523/how-to-go-from-solidity-ast-back-to-source-code) recommends [this library](https://github.com/chao-peng/SIF).

Comment (pavelnikolov) [reply contracts/bindings/scripts/commenttypes.go:77]: I didn't realize that this was Go code. Here is [an example](https://go.dev/src/go/ast/example_test.go).

**Pull request merged**